### PR TITLE
CORCI-608: Create a customized config file for pylint

### PR DESCRIPTION
**Pull request by skirvan44**
Signed-off-by: Scott Kirvan <scott.kirvan@intel.com>

### CODE REVIEW

File: pylintrc [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+# you want to run only the similarities checker, you can use "--disable=all
+# --enable=similarities". If you want to run only the classes checker, but have
+# no Warning level messages displayed, use"--disable=all --enable=classes
+# --disable=W"
+disable=import-star-module-level,old-octal-literal,oct-method,print-statement,unpacking-in-except,parameter-unpacking,backtick,old-raise-syntax,old-ne-operator,long-suffix,dict-view-method,dict-iter-method,metaclass-assignment,next-method-called,raising-string,indexing-exception,raw_input-builtin,long-builtin,file-builtin,execfile-builtin,coerce-builtin,cmp-builtin,buffer-builtin,basestring-builtin,apply-builtin,filter-builtin-not-iterating,using-cmp-argument,useless-suppression,range-builtin-not-iterating,suppressed-message,no-absolute-import,old-division,cmp-method,reload-builtin,zip-builtin-not-iterating,intern-builtin,unichr-builtin,reduce-builtin,standarderror-builtin,unicode-builtin,xrange-builtin,coerce-method,delslice-method,getslice-method,setslice-method,input-builtin,round-builtin,hex-method,nonzero-method,map-builtin-not-iterating,import-error,wrong-import-position
```
Comment (brianjmurrell) [inline pylintrc]: That seems like it's disabling a lot of checks.  Do we really want to disable all of those?

File: pylintrc [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+
+# Template used to display messages. This is a python new-style format string
+# used to format the message information. See doc for all details
+#msg-template=
+
+
+[FORMAT]
+
+# Maximum number of characters on a single line.
+max-line-length=100
```
Comment (brianjmurrell) [inline pylintrc]: I thought as a group we agreed on 80.

File: pylintrc (hunk 1 above)
Comment (jolivier23) [reply pylintrc]: I think it would better to follow pylint defaults where they are not unreasonable

Comment (skirvan44) [reply pylintrc]: Those are the defaults except the 2 at the end.  I asked pylint to generate the file based on defaults and then I just changed a couple things per discussion at hudson.

File: pylintrc (hunk 2 above)
Comment (skirvan44) [reply pylintrc]: That is pylint default but I'm fine changing it to 80.  I think other checkers are catching it if its over 80.

**Comment by skirvan44** [comment]
As I said elsewhere, these are the defaults except I modified a couple things like how many arguments a function can have, how many variables you can declare in a function etc. per discussion with team.

**Comment by brianjmurrell** [comment]
> As I said elsewhere, these are the defaults except I modified a couple things like how many arguments a function can have, how many variables you can declare in a function etc. per discussion with team.

So, I think the technique of asking `pylint` to generate a `pylintrc` of all of the (i.e. current) defaults and then using that file is fine if the intention is to always use today's defaults, forever forward.

But is that really what we want to do or do we want to allow pylint to introduce new requirements on our code as PEP-8 evolves and pylint evolves with it?

As a point of contrast, I have a local `pylintrc` file for DAOS which only has in it:

```
[MASTER]
init-hook='sys.path.append("src/tests/ftest/util/apricot/"); sys.path.append("src/utils/py"); sys.path.append("src/tests/ftest/util")'

[MESSAGES CONTROL]

disable=too-many-branches,
        too-many-statements,
        too-many-instance-attributes,
        too-many-locals,
        too-few-public-methods,
        broad-except,
        fixme,
        global-statement
enable=line-too-long

[BASIC]
no-docstring-rgx=^_|^setUp$|^tearDown$
good-names=setUp,tearDown
```

I'm definitely not suggesting we use those settings, but just offer the file as example of how to modify the defaults I am interested in overriding but also allow the defaults to evolve as pylint and PEP-8 evolve.

**Comment by skirvan44** [comment]
Okay, I'm fine with that approach.  I wasn't sure if just having a rc file overrode all the defaults but it sounds like if I don't explicitly change the default then it is still present.  I'll try it and update if that works.

**Pull request closed without merging**